apps/api/test/functional/trial-providers.spec.ts

```diff
@@ -0,0 +1,141 @@
+import { ProviderAttributeSignature, ProviderSnapshot } from "@akashnetwork/database/dbSchemas/akash";
+import subDays from "date-fns/subDays";
+import mcache from "memory-cache";
+import nock from "nock";
+
+import { app, initDb } from "@src/app";
+import { closeConnections } from "@src/db/dbConnection";
+import { AUDITOR, TRIAL_ATTRIBUTE, TRIAL_REGISTERED_ATTRIBUTE } from "@src/deployment/config/provider.config";
+
+const createProviderSnapshot = async (
+  owner: string,
+  daysAgo: number,
+  availableCPU: number,
+  availableGPU: number,
+  availableMemory: number,
+  availablePersistentStorage: number,
+  availableEphemeralStorage: number
+) => {
+  await ProviderSnapshot.create({
+    owner,
+    checkDate: subDays(new Date("2025-01-01T00:00:00.000Z"), daysAgo),
+    isOnline: true,
+    availableCPU,
+    availableGPU,
+    availableMemory,
+    availablePersistentStorage,
+    availableEphemeralStorage
+  });
+};
+
+describe("Trial Providers", () => {
+  const provider1 = "akash1rk090a6mq9gvm0h6ljf8kz8mrxglwwxsk4srxh";
+  const provider2 = "akash1d4fletej4cwn9x8jzpzmnk6zkqeh90ejjskpmu";
+
+  beforeAll(async () => {
+    await expect.getState().dbService.copyIndexerTables(["providerAttributeSignature", "provider"]);
```

Comment:
as I mentioned in a different comment, this wouldn't work as it relies on a db existence which is not imported in CI.
```suggestion
```

You can use this once merged to pre-create providers so that seeders below work https://github.com/akash-network/console/pull/1257/files#diff-0107cc7c80b1a3372b2d16da819414c1856899b7866b8ddc5a663489cacc9932R5